ksql-engine/src/test/java/io/confluent/ksql/SchemaTranslationTest.java

```diff
@@ -0,0 +1,193 @@
+            topic,
+            "test-key",
+            generator.generate(),
+            0,
+            null
+        )
+    ).collect(Collectors.toList());
+  }
+
+  @SuppressWarnings("unchecked")
+  private static List<Record> getOutputRecords(
+      final Topic topic, final List<Record> inputRecords,
+      final org.apache.avro.Schema avroSchema) {
+    return inputRecords.stream()
+        .map(
+            r -> new Record(
+                topic,
+                "test-key",
+                avroToValueSpec(r.value(), avroSchema, true),
+                0,
+                null
+            ))
+        .collect(Collectors.toList());
+
+  }
+
+  private static Object loadRecordSpec(final JsonNode node) {
+    try {
+      return objectMapper.readValue(
+          objectMapper.writeValueAsString(node), Map.class);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  private static List<Record> loadRecords(final Topic topic, final JsonNode node) {
+    final List<Record> records = new LinkedList<>();
+    node.forEach(
+        child -> records.add(
+            new Record(
+                topic,
+                "test-key",
+                loadRecordSpec(child),
+                0,
+                null
+            )
+        )
+    );
+    return records;
+  }
+
+  private static Query loadTest(final JsonNode node) {
+    final JsonNode schemaNode = node.get("schema");
+    final String schemaString;
+    try {
+      schemaString = new ObjectMapper().writeValueAsString(schemaNode);
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException(e);
+    }
+
+    final org.apache.avro.Schema.Parser parser = new org.apache.avro.Schema.Parser();
+    final org.apache.avro.Schema avroSchema = parser.parse(schemaString);
+
+    final Topic srcTopic;
+    final Topic outputTopic
+        = new Topic(OUTPUT_TOPIC_NAME, null, new ValueSpecAvroSerdeSupplier());
+    final List<Record> inputRecords;
+    final List<Record> outputRecords;
+    if (node.has("input_records")) {
+      srcTopic = new Topic(TOPIC_NAME, avroSchema, new ValueSpecAvroSerdeSupplier());
+      inputRecords = loadRecords(srcTopic, node.get("input_records"));
+      outputRecords = loadRecords(outputTopic, node.get("output_records"));
+    } else {
+      srcTopic = new Topic(TOPIC_NAME, avroSchema, new AvroSerdeSupplier());
+      inputRecords = generateInputRecords(srcTopic, avroSchema);
+      outputRecords = getOutputRecords(outputTopic, inputRecords, avroSchema);
+    }
+
+    final String ddlStatement =
+        String.format(
+            "CREATE STREAM %s WITH (KAFKA_TOPIC='%s', VALUE_FORMAT='AVRO');",
+            TOPIC_NAME, TOPIC_NAME);
+    final String csasStatement = avroSchema.getFields()
+        .stream()
+        .map(Schema.Field::name)
+        .collect(
+            Collectors.joining(
+                ", ",
+                "CREATE STREAM TEST_OUTPUT AS SELECT ",
```

Comment:
I am just brain storming here, but would it make sense to use the fields in more contexts than just a simple projection? For example, as part of an argument to a UDF, or in an expression? Would these additional scenarios actually test something additional?

Reply:
Also, how would this work with accessing fields within structs using dot notation? It seems to me that as is, this would just project the top level field.